src/main/kotlin/org/rust/ide/intentions/SimplifyBooleanExpressionIntention.kt

```diff
@@ -0,0 +1,122 @@
+package org.rust.ide.intentions
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.tree.IElementType
+import org.rust.ide.formatter.impl.UNARY_OPS
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.RustTokenElementTypes.*
+import org.rust.lang.core.psi.util.elementType
+import org.rust.lang.core.psi.util.parentOfType
+
+class SimplifyBooleanExpressionIntention : RustElementBaseIntentionAction<SimplifyBooleanExpressionIntention.Context>() {
+    override fun getText() = "Simplify boolean expression"
+    override fun getFamilyName() = text
+
+    data class Context(
+        val expr: RustExprElement
```

Comment:
I thought that I would add other types support and I would need some additional meta-information. But for now `expr` is enough, so I will eliminate useless data class.